SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -363,6 +362,146 @@ void InterestFactory::AddServerGameplayDebuggerCategoryReplicatorActorInterest(I
 }
 #endif
 
+bool UnrealServerInterestFactory::CreateClientInterestDiff(const APlayerController* PlayerController,
+														   ChangeInterestRequest& ChangeInterestRequestData, const bool bOverwrite) const
+{
+	USpatialNetConnection* NetConnection =
+		(PlayerController != nullptr) ? Cast<USpatialNetConnection>(PlayerController->NetConnection) : nullptr;
+	if (!ensure(GetDefault<USpatialGDKSettings>()->bUseClientEntityInterestQueries)
+		|| !ensureMsgf(PlayerController != nullptr, TEXT("Only PCs can use client entity interest: %s"), *GetNameSafe(PlayerController))
+		|| !ensureMsgf(NetConnection != nullptr, TEXT("Failed to retieve USpatialNetConnection from controller: %s"),
+					   *GetNameSafe(PlayerController)))
+	{
+		return false;
+	}
+
+	TArray<Worker_EntityId> ClientInterestedEntities = GetClientInterestedEntityIds(PlayerController);
+
+	USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(PlayerController->GetNetDriver());
+
+	// MAKE DEBUGGING EASIER, REVERT ME
+#if WITH_EDITOR
+	ClientInterestedEntities.Sort();
+#endif
+
+	ChangeInterestRequestData.Clear();
+
+	{
+		// Add non-auth interest
+		TSet<Worker_EntityId_Key> FullInterested;
+		FullInterested.Reserve(ClientInterestedEntities.Num());
+		for (auto EntityId : ClientInterestedEntities)
+		{
+			FullInterested.Add(EntityId);
+		}
+
+		TSet<Worker_EntityId_Key> Add, Remove;
+		if (bOverwrite)
+		{
+			Add = FullInterested;
+			Remove.Empty();
```

Comment:
I assume TSet is empty initialised? is this a code clarity thing?

Reply:
Yep code clarity